src/app/prevention/strides-against/strides-against.component.ts

```diff
@@ -0,0 +1,239 @@
+import { Component, OnInit, ViewChild } from '@angular/core';
+import { LanguageService } from '../../services/language.service';
+import { MatDialog } from '@angular/material';
+import { DialogComponent } from './dialog/dialog.component';
+import { Observable } from 'rxjs/Observable';
+import { SharedDataService } from '../../services/shared.data.service';
+import swal from 'sweetalert2';
+import { Router } from '@angular/router';
+import { BackgroundConfig, ManConfig, ObjectsConfig, positionConfig } from './game-config';
+import { DashboardService } from '../../services/dashboard.service';
+import { PositionElement } from './models/elements';
+import { Item } from './models/Item';
+import { InfokitService } from '../../services/infokit.service';
+
+export class PaperConfig {
+  imageIndex: number;
+  imagePositionIndex: number;
+}
+
+@Component({
+  selector: 'app-strides-against',
+  templateUrl: './strides-against.component.html',
+  styleUrls: ['./strides-against.component.scss']
+})
+export class StridesAgainstComponent implements OnInit {
+  // Initialising game area
+  @ViewChild('gameArea') gameArea;
+  paper: RaphaelPaper;
+  man: RaphaelElement;
+
+  language: any;
+  alerts: any;
+  pos = 0;
+  question: string;
+  result: any;
+
+  completed = false;
+  activityComplete = false;
+
+  // Stores the objects
+  elements: RaphaelElement[] = [];
+
+  answer: boolean;
+
+  // Configs:
+  BackgroundConfig = BackgroundConfig;
+  ObjectsConfig = ObjectsConfig;
+  ManConfig = ManConfig;
+  randomPosition: Array<PositionElement> = []; // After randomizing the original position config
+  positionConfig = positionConfig;
+
+  randomObjectConfig: Array<Item> = []; // After randomizing the original  objects config
+  paperConfig: Array<PaperConfig> = [];
+  sortedPosition: Array<PositionElement>;
+
+  // Changing this would change the no. of objects in the game
+  countImage = BackgroundConfig.foreObjectsCount;
+  xMoveSign = 1; // Positive for moving in positive direction and vice versa
+
+
+  constructor(public languageService: LanguageService, public dialog: MatDialog, public sharedDataService: SharedDataService,
+              public router: Router, public dashboardService: DashboardService, private infokitService: InfokitService) {
+  }
+
+  /**
+   * Returns a random ordered Array
+   * @param  {array} array which needs to be randomized
+   */
+  randomArray(array) {
+    for (let i = array.length - 1; i > 0; i--) {
+      const j = Math.floor(Math.random() * (i + 1));
+      [array[i], array[j]] = [array[j], array[i]];
+    }
+    return array;
+  }
+
+  /**
+   * Called after the man has moved linearly to the desired position
+   */
+  handleQuestion(): void {
+    let currentImage = 0;
+    let index = this.pos - 1; // Finding which object eventually landed at the first position and so on...
```

Comment:
What is this for? Won't index be -1 when this.pos is 0?

Reply:
@Raxerz Actually, the handleQuestion( ) is called after incrementing the 'pos' variable first. That way, it would never be negative.

Reply:
Oh okay. I think I overlooked that. Thanks for clarifying.